addons/tutorials-button/addon.json

```diff
@@ -1,21 +1,139 @@
   "credits": [
+    {
+      "name": "philipp2007",
+      "link": "https://ggigabyte.repl.co/",
+      "note": "update, menus, compact user dropdown",
+      "id": "update"
+    },
+    {
+      "name": "CST1229",
+      "link": "https://github.com/CST1229/",
+      "note": "compact user dropdown",
+      "id": "dropdown"
+    },
     {
       "name": "lisa_wolfgang",
-      "link": "https://scratch.mit.edu/users/lisa_wolfgang/"
+      "link": "https://scratch.mit.edu/users/lisa_wolfgang/",
+      "note": "hide tutorials button",
+      "id": "tutorials-button"
     }
   ],
-  "userstyles": [
+  "tags": ["editor", "editorMenuBar", "featured"],
+  "settings": [
+    {
+      "name": "Menus",
+      "id": "menu-labels",
+      "type": "select",
+      "potentialValues": [
+        {
+          "name": "Show icons and text",
+          "id": "both"
+        },
+        {
+          "name": "Only show icons",
+          "id": "icons"
+        }
+      ],
+      "default": "both"
+    },
+    {
+      "name": "Tutorials button",
+      "id": "tutorials-button",
+      "type": "select",
+      "potentialValues": [
+        {
+          "name": "Show icon and text",
+          "id": "both"
+        },
+        {
+          "name": "Only show icon",
+          "id": "icon"
+        },
+        {
+          "name": "Hide",
+          "id": "hide"
+        }
+      ],
+      "default": "both"
+    },
+    {
+      "name": "Hide My Stuff button",
+      "id": "my-stuff",
+      "type": "boolean",
+      "default": false
+    },
     {
-      "url": "userstyle.css",
+      "name": "Compact user dropdown",
+      "id": "compact-username",
+      "type": "boolean",
+      "default": false
+    }
+  ],
+  "userscripts": [
+    {
+      "url": "username.js",
       "matches": ["projects"]
     }
   ],
-  "versionAdded": "1.22.0",
-  "dynamicEnable": true,
+  "userstyles": [
+    {
+      "url": "my-stuff-hide.css",
+      "matches": ["projects"],
+      "if": {
+        "settings": {
+          "my-stuff": true
+        }
+      }
+    },
+    {
+      "url": "username.css",
+      "matches": ["projects"],
+      "if": {
+        "settings": {
+          "compact-username": true
+        }
+      }
+    },
+    {
+      "url": "menu-icons.css",
+      "matches": ["projects"],
+      "if": {
+        "settings": {
+          "menu-labels": "icons"
+        }
+      }
+    },
+    {
+      "url": "tutorials-icon.css",
+      "matches": ["projects"],
+      "if": {
+        "settings": {
+          "tutorials-button": "icon"
+        }
+      }
+    },
+    {
+      "url": "tutorials-hide.css",
+      "matches": ["projects"],
+      "if": {
+        "settings": {
+          "tutorials-button": "hide"
+        }
+      }
+    }
+  ],
   "dynamicDisable": true,
-  "enabledByDefault": false
+  "dynamicEnable": true,
+  "updateUserstylesOnSettingsChange": true,
+  "enabledByDefault": false,
+  "versionAdded": "1.22.0",
```

Comment:
Should we consider this a new addon or not?

Reply:
That update from "Show Discuss button" to "Custom navbar" wasn't considered as new addon. They both have the same addon ID, so...

Reply:
I'd encourage you to change the addon ID, as we'll need to add settings migration code nevertheless

Reply:
So... basically revert commit [451347e](https://github.com/ScratchAddons/ScratchAddons/pull/6297/commits/451347e8d754fb439574566f42ebf6a80196f891)?

Reply:
Yep

Reply:
Done!